lib/sales_analyst.rb

```diff
@@ -0,0 +1,24 @@
+class SalesAnalyst
+
+  attr_reader :items,
+              :merchants
+
+  def initialize(items, merchants)
+    @merchants = merchants
+    @items = items
+  end
+
+  def merch_item_creation(items, merchants)
+    return_hash = {}
+    merchants.all.each do |merchant|
+    items.all.each do |item|
+      if merchant.id == item.merchant.id
+        if return_hash[merchant].nil?
+          return_hash[merchant.name] = [item]
+        else
+          return_hash[merchant.name] += [item]
+        end
+      end
```

Comment:
You can actually do this in a really cool way!

`items.all.each do |item|
	return_hash[merchant.name] ||= []
	return_hash[merchant.name] << item
end`

What ||= (or equals) does is really cool and helps break this down further; it essentially checks if the hash key you call (in this case, return_hash[merchant.name]) returns a nil, and if it does it reads the rest of the line (thus setting the key equal to an empty array). If the hash key you call *doesn't* return a nil, it skips the rest of the line after the ||= and moves down to the next line, meaning that you simply shovel in the item rather than also creating a new key value pair on top of it.